### ISSUE TITLE
`tf.random.set_seed` lack input validation

### ISSUE THREAD

**Issue by ArrowIntoTheSky** [tensorflow/tensorflow#55185]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Y
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.7.0
- Python version: 3.8
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source):N/A
- CUDA/cuDNN version: N/A
- GPU model and memory: N/A

**Standalone code to reproduce the issue**
```
import tensorflow as tf
seed = None
tf.random.set_seed(seed, )
a = tf.random.uniform([1, 2])
tf.random.set_seed(seed, )
b = tf.random.uniform([1, 2])
tf.debugging.assert_near(a, b) # InvalidArgumentError
```

**Describe the current behavior**
[`tf.random.set_seed`](https://www.tensorflow.org/api_docs/python/tf/random/set_seed) should be called with an integer, however, calling `tf.random.set_seed(None)` will not result in error, and it cannot guarantee reproducibility : `a` is not equal to `b`.

**Describe the expected behavior**
`tf.random.set_seed` should check properly that `seed` is an integer, and raise an error if `seed` is `None`.

**Comment by jvishnuvardhan** [comment]
@ArrowIntoTheSky Thanks for raising the issue. Added internal CL to check the input seed value and throw an error if it is `None`. Thanks

**Comment by mohantym** [comment]
Hi @ArrowIntoTheSky ! This issue has been fixed in 2.8 version now. Can we move this to closed status now?
